### PR TITLE
APPS-3489: Add propel schema

### PR DESCRIPTION
**Pull request by glazin-logema**
Ticket: https://spryker.atlassian.net/browse/APPS-3489

### CODE REVIEW

File: config/spryk/spryks/Spryker/Zed/Business/AddCrudFacade.yml [hunk 1]
```diff
@@ -22,3 +22,4 @@ preSpryks:
     - AddModule
 
 postSpryks:
+    - AddZedPersistencePropelSchemaProperty
```
Comment (stereomon) [inline config/spryk/spryks/Spryker/Zed/Business/AddCrudFacade.yml:25]: You also need to add the arguments. This Spryk should create an XML file with the defined entities in https://spryker.atlassian.net/wiki/spaces/AOP/pages/3224371207/WIP+APPS-3483+CRUD+Facade+function+RFC+to+Spryk#Domain-model-diagram

Comment (dereuromark) [reply config/spryk/spryks/Spryker/Zed/Business/AddCrudFacade.yml:25]: Do we need to leak the Propel part here everywhere?
I find it clear enough as persistence schema to be honest

> AddZedPersistenceSchemaProperty

File: config/spryk/spryks/Spryker/Zed/Persistence/Propel/Schema/AddZedPersistencePropelSchemaProperty.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+spryk: schema
+description: "Adds a propel schema to *module*.schema.xml in the Zed application."
```
Comment (dereuromark) [inline config/spryk/spryks/Spryker/Zed/Persistence/Propel/Schema/AddZedPersistencePropelSchemaProperty.yml:2]: Same here
```suggestion
description: "Adds a Persistence schema to *module*.schema.xml in the Zed application."
```

File: config/spryk/spryks/Spryker/Zed/Persistence/Propel/Schema/AddZedPersistencePropelSchemaProperty.yml [hunk 3]
```diff
@@ -11,18 +10,25 @@ arguments:
...
-    name:
-        inherit: true
-
     propertyName: ~
 
     propertyType: ~
 
-    singular:
-        default: false
+    required: ~
```
Comment (stereomon) [inline config/spryk/spryks/Spryker/Zed/Persistence/Propel/Schema/AddZedPersistencePropelSchemaProperty.yml:17]: Add useful defaults to those arguments. You can see what happens if we don't have good default values in the AddCrudFacade spryk, many arguments need to be set there to be just empty.

File: config/spryk/spryks/Spryker/Zed/Persistence/Propel/Schema/AddZedPersistencePropelSchemaProperty.yml [hunk 4]
```diff
@@ -11,18 +10,25 @@ arguments:
...
-        value: "src/{{ organization }}/Zed/{{ module }}/Persistence/Propel/Schema/{{ module | underscored }}.schema.xml"
+        value: "src/{{ organization }}/Zed/{{ module }}/Persistence/Propel/Schema/spy_{{ module | underscored }}.schema.xml"
 
 preSpryks:
-    - AddZedPersistencePropelSchema
+    - AddZedPersistencePropelSchemaTable:
+          arguments:
+              tableName:
+                  value: "spy_{{ module | underscored }}"
+              targetModule:
```
Comment (stereomon) [inline config/spryk/spryks/Spryker/Zed/Persistence/Propel/Schema/AddZedPersistencePropelSchemaProperty.yml:33]: Can't you re-use `module`?

Comment (glazin-logema) [reply config/spryk/spryks/Spryker/Zed/Persistence/Propel/Schema/AddZedPersistencePropelSchemaProperty.yml:40]: If I remove targetModule from here and replace it with module or something else, everything breaks. Here now the module is passed to the value, I guess I didn't get your point completely

Comment (stereomon) [reply config/spryk/spryks/Spryker/Zed/Persistence/Propel/Schema/AddZedPersistencePropelSchemaProperty.yml:40]: If it only works with `targetModule` I'm fine.

**Review by stereomon — APPROVED** [review]

**Pull request merged**